x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml

```diff
@@ -15,3 +15,46 @@ template:
         type: keyword
         time_series_dimension: true
         ignore_above: 1024
+    dynamic_templates:
+      - histogram:
+          mapping:
+            type: histogram
+            ignore_malformed: true
+      - counter_long:
+          mapping:
+            type: long
+            time_series_metric: counter
+            ignore_malformed: true
+      - gauge_long:
+          mapping:
+            type: long
+            time_series_metric: gauge
+            ignore_malformed: true
+      - counter_double:
+          mapping:
+            type: double
+            time_series_metric: counter
+            ignore_malformed: true
+      - gauge_double:
+          mapping:
+            type: double
+            time_series_metric: gauge
+            ignore_malformed: true
+      - summary_gauge:
+          mapping:
+            type: aggregate_metric_double
+            time_series_metric: gauge
+            ignore_malformed: true
+      - summary_counter:
+          mapping:
+            type: aggregate_metric_double
+            time_series_metric: counter
+            ignore_malformed: true
+      - histogram_metrics:
+          mapping:
+            type: histogram
+      - summary_metrics:
+          mapping:
+            type: aggregate_metric_double
+            metrics: sum, value_count
+            default_metric: value_count
```

Comment:
These 2 are duplicates from `metrics@mappings.json` - they were already there before, so I also left them in `metrics@mappings.json`. Do we actually need these 2 here for OTel?

Reply:
In https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/elasticsearchexporter/model.go, I could not find usage of the following dynamic templates. Thus, I think they can be removed.

* summary_gauge
* summary_counter
* histogram_metrics

Reply:
Will they be useful in any scenarios? If there is a chance that we need it, might be useful to keep these if a future version of es exporter need to work with 8.16 ES. Just asking if see how likely we need them.

Reply:
After looking into this, I'd say it's safe to remove all 3.

- `histogram_metrics`: That is covered by our `histogram` template, which is the same plus it adds  `ignore_malformed: true`. Without `ignore_malformed: true`, I doubt we'd want to re-use this. `histogram_metrics` comes from the old core templates.
- `summary_counter`: This is covered by `counter_long` and `counter_double`, the difference is that this one sets `type: aggregate_metric_double` instead of double and long. By looking at [docs for counters in OTel](https://opentelemetry.io/docs/specs/otel/metrics/api/#counter-creation) I doubt we'd want to use `aggregate_metric_double` - that does not match the OTel counters.
- `summary_gauge`: same as above, and this is covered by `gauge_long` and `gauge_double`.

So with that, I don't think we'd want to use those dynamic templates for OTel later. -> I'd be for just removing.

Any thoughts? Maybe something I'm overlooking?

Reply:
sgtm